cmd/dockerd/docker.go

```diff
@@ -23,6 +25,16 @@ func main() {
 		return
 	}
 
+	// Fixes https://github.com/docker/docker/issues/19728
+	// Ignore SIGPIPE events, that are generated by systemd when journald is restarted while
+	// the docker daemon is not restarted and also running under systemd.
+	go func() {
+		c := make(chan os.Signal)
+		signal.Notify(c, os.Signal(syscall.SIGPIPE))
+		for range c {
```

Comment:
@LK4D4 I didn't log any message to handle the use case where the logging is going to stdout/stderr.  Logging here in that case would create an infinite loop.